docs/proposals/04-new-core-gardener-cloud-apis.md

```diff
@@ -0,0 +1,716 @@
+Meanwhile, we have progressed a lot and are about to remove the [`CloudBotanist` interface](https://github.com/gardener/gardener/blob/master/pkg/operation/cloudbotanist/types.go) entirely.
+The only missing aspect that will allow providers to really maintain their code out of the core is to design new APIs.
+
+This proposal describes how the new `Shoot`, `Seed` etc. APIs will be re-designed to cope with the changes made with extensibility.
+We already have the new `core.gardener.cloud/v1alpha1` API group that will be the new default soon.
+
+## Motivation
+
+We want to allow providers to individually maintain their specific knowledge without the necessity to touch the Gardener core code.
+In order to achieve the same, we have to provide proper APIs.
+
+### Goals
+
+* Provide proper APIs to allow providers maintaining their code outside of the core codebase.
+* Do not complicate the APIs for end-users such that they can easily create, delete, and maintain shoot clusters.
+
+### Non-Goals
+
+* Let's try to not split everything up into too many different resources. Instead, let's try to keep all relevant information in the same resources when possible/appropriate.
+
+## Proposal
+
+In GEP-1 we already have proposed a first version for new `CloudProfile` and `Shoot` resources.
+In order to deprecate the existing/old `garden.sapcloud.io/v1beta1` API group (and remove it, eventually) we should move all existing resources to the new `core.gardener.cloud/v1alpha1` API group.
+
+### `ProviderProfile` resource
+
+Special note: The `CloudProfile` will be renamed to `ProviderProfile`.
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: backup-secret
+  namespace: garden
+type: Opaque
+data:
+  # <some-provider-specific data keys>
+  # https://github.com/gardener/gardener-extensions/blob/master/controllers/provider-alicloud/example/30-backupbucket.yaml#L9-L11
+  # https://github.com/gardener/gardener-extensions/blob/master/controllers/provider-aws/example/30-infrastructure.yaml#L9-L10
+  # https://github.com/gardener/gardener-extensions/blob/master/controllers/provider-azure/example/30-backupbucket.yaml#L9-L10
+  # https://github.com/gardener/gardener-extensions/blob/master/controllers/provider-gcp/example/30-backupbucket.yaml#L9
+  # https://github.com/gardener/gardener-extensions/blob/master/controllers/provider-openstack/example/30-backupbucket.yaml#L9-L13
+
+---
+apiVersion: core.gardener.cloud/v1alpha1
+kind: ProviderProfile
+metadata:
+  name: providerprofile1
+spec:
+  type: <some-provider-name> # {aws,azure,gcp,...}
+# Optional list of labels on `Seed` resources that marks those seeds whose shoots may use this provider profile.
+# An empty list means that all seeds of the same provider type are supported.
+# This is useful for environments that are of the same type (like openstack) but may have different "instances"/landscapes.
+# seedSelector:
+#   matchLabels:
+#     foo: bar
+  kubernetes:
+    versions:
```

Comment:
How about to have an `expirationDate` per version, just like for the machine images?

Reply:
Ok, this is already in discussion in #1325. So I guess you will update this GEP once that issue is finalized.

Reply:
Yes, I'll add it now. I guess it's pretty clear how that should look like, thanks.